File: CodeForces/901-1000/932/A/932A.rb
# Problem 932A: Palindromic Supersequence
# https://codeforces.com/contest/932/problem/A
# Executed with: ruby 3.1.2p20 (2022-04-12 revision 4491bb740a) [x86_64-linux-gnu]
# Linux version: Linux kernel 6.1.0-10-amd64
# Submitted on: October 23rd, 2025
# Codeforces language used: Ruby 3.2.2

w = gets.chomp
w_back = w.dup
w_back.reverse!

puts w + w_back

File: CodeForces/901-1000/932/A/932A.pl
#!/usr/bin/perl

#! Problem 932A: Palindromic Supersequence
#! https://codeforces.com/contest/932/problem/A
#! Executed with: perl 5, version 40, subversion 1 (v5.40.1) built for aarch64-linux-gnu-thread-multi
#! Linux version: Linux kernel 6.17.0-6-generic
#! Submitted on: October 27th, 2025
#! Codeforces language used: Perl 5.20.1

use strict;
use warnings;

chomp(my $n = <STDIN>);
print $n, scalar reverse($n), "\n";
